### PR TITLE
Feature/add express checkout app

### PR DESCRIPTION
**Pull request by sasha0**

### CODE REVIEW

File: paypal/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-VERSION = '2.0.0'
+VERSION = '3.0.0'
```
Comment (sasha0) [inline paypal/__init__.py:1]: Let's not bump it yet.

File: paypal/express_checkout/models.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+from django.db import models
+
+
+class ExpressCheckoutTransaction(models.Model):
```
Comment (sasha0) [inline paypal/express_checkout/models.py:4]: Please add a docstring, explaining the purpose of this model (e.g. dedicated to store transactions for "express checkout sdk".

File: paypal/express_checkout/views.py [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+        return super().get(request, *args, **kwargs)
+
+    def get_redirect_url(self, **kwargs):
+        messages.error(self.request, _('PayPal transaction cancelled'))
+        return reverse('basket:summary')
+
+
+# Upgrading notes: when we drop support for Oscar 0.6, this class can be
+# refactored to pass variables around more explicitly (instead of assigning
+# things to self so they are accessible in a later method).
```
Comment (sasha0) [inline paypal/express_checkout/views.py:140]: This is not necessary nowadays.

File: sandbox/settings.py [hunk 4]
```diff
@@ -305,8 +310,7 @@
 
 # Taken from PayPal's documentation - these should always work in the sandbox
 PAYPAL_SANDBOX_MODE = True
-PAYPAL_CALLBACK_HTTPS = False
```
Comment (sasha0) [inline sandbox/settings.py:308]: Let's leave it for now, for backwards compatibility and replace in 3.0 (alongside with other breaking changes).

File: sandbox/settings.py [hunk 5]
```diff
@@ -305,8 +310,7 @@
...
-PAYPAL_API_VERSION = '119'
```
Comment (sasha0) [inline sandbox/settings.py:309]: This one is used by express gateway, so let's leave it for now (in 2.x).

File: pytest.ini [hunk 6]
```diff
@@ -1,2 +1,3 @@
 [pytest]
+testpaths = tests
```
Comment (sasha0) [inline pytest.ini:2]: Can we address this in the separate PR please?

File: tests/urls.py [hunk 7]
```diff
@@ -2,7 +2,7 @@
 from django.conf.urls import include, url
 from django.conf.urls.i18n import i18n_patterns
 
-from paypal.express.urls import base_patterns, buyer_pays_on_paypal_patterns, buyer_pays_on_website_patterns
+from paypal.express_checkout.urls import base_patterns, buyer_pays_on_paypal_patterns, buyer_pays_on_website_patterns
```
Comment (sasha0) [inline tests/urls.py:5]: Why won't we leave both of them for now?

**Review by sasha0 — COMMENTED** [review]
Let's extract it to the separate PR. Actually, I'll do that by myself.

File: setup.py [hunk 8]
```diff
@@ -18,6 +18,7 @@
     include_package_data=True,
     install_requires=[
         'django>=2.2,<2.3',
+        'paypal-checkout-serversdk>=1.0.1'
```
Comment (sasha0) [inline setup.py:21]: Please add missing trailing comma.

**Review by sasha0 — COMMENTED** [review]
Good findings - I already extracted it to the separate PR.

File: pytest.ini (hunk 6 above)
Comment (samitnuk) [reply pytest.ini:2]: Done in https://github.com/django-oscar/django-oscar-paypal/pull/251

File: paypal/express_checkout/models.py (hunk 2 above)
Comment (samitnuk) [reply paypal/express_checkout/models.py:4]: Done.

**Pull request merged**